src/core/lib/surface/channel.cc

```diff
@@ -58,21 +58,20 @@
 static void destroy_channel(void* arg, grpc_error_handle error);
 
 grpc_channel* grpc_channel_create_with_builder(
-    grpc_channel_stack_builder* builder,
+    grpc_core::ChannelStackBuilder* builder,
     grpc_channel_stack_type channel_stack_type, grpc_error_handle* error) {
-  std::string target = grpc_channel_stack_builder_get_target(builder);
-  grpc_channel_args* args = grpc_channel_args_copy(
-      grpc_channel_stack_builder_get_channel_arguments(builder));
+  std::string target(builder->target());
```

Comment:
I think we no longer need to make these copies of the target or channel args.  They were previously necessary because `grpc_channel_stack_builder_finish()` would destroy the builder, but that's no longer the case, so we should be able to continue to access the copies inside of the builder where we need them below.